src/EventStore.Client/EventStoreClientSettings.ConnectionString.cs

```diff
@@ -204,21 +220,45 @@ private static class ConnectionStringParser {
 					settings.ConnectivitySettings.TlsVerifyCert = (bool)tlsVerifyCert;
 				}
 
+				if (typedOptions.TryGetValue(TlsCaFile, out var tlsCaFile)) {
+					var tlsCaFilePath = Path.GetFullPath((string)tlsCaFile);
+					if (!string.IsNullOrEmpty(tlsCaFilePath) && !File.Exists(tlsCaFilePath)) {
+						throw new FileNotFoundException($"Failed to load certificate. File was not found.");
+					}
+
+					try {
+						using var x509Certificate2 = new X509Certificate2(tlsCaFilePath);
+						settings.ConnectivitySettings.TlsCaFile = tlsCaFilePath;
+					} catch (CryptographicException) {
+						throw new Exception("Failed to load certificate. Invalid file format.");
+					}
+				}
+
 				settings.CreateHttpMessageHandler = CreateDefaultHandler;
```

Comment:
I think we may also need to handle the client certificate in `EventStore.Client.ChannelFactory` since it has its own 'fallback' `CreateDefaultHandler` that diverges from the logic here.

Reply:
Done 👍